### PR TITLE
[blockly] add more migration info

### PR DESCRIPTION
**Pull request by stefan-hoehn**
Adds more information regarding migrating blocklies from openHAB 3 to openHAB 4

### CODE REVIEW

File: configuration/blockly/index.md [hunk 1]
```diff
@@ -51,13 +51,43 @@ Using this library you can not only create much simpler code, it also allows **n
...
+#### Migrating Blockly rules to openHAB 4 is easy
 
-- From openHAB 4 on, the default script engine is GraalJS when Blockly creates new scripts
-- If you still want to run Blockly rules that were created in openHAB3 without changing them (see below) you can install an automation plugin for Nashorn which could provide backwards compatibility until you have converted all rules.
-After installation both old code and new code can run in openHAB4.
-- **To convert a rule that was created in openHAB 3 (NashornJS) to a new rule based on GraalJS for openHAB 4 simply open the rule once in openHAB 4 and save it - that's it.**
-- After all rules have been converted, please uninstall the Nashorn addon to save memory.
+From openHAB 4 on, the default script engine is GraalJS when Blockly creates new scripts.
+From a technical perspective a rule internally holds a so-called MIME-type that tells openHAB how the generated JavaScript language has to be interpreted.
+The (default) MIME-type application/javascript in openHAB 3 runs the rule with NashornJS, while this same MIME-type will run the Blocky rule with GraalJS.
```
Comment (florian-h05) [inline configuration/blockly/index.md:58]:
```suggestion
The (default) MIME-type `application/javascript` in openHAB 3 runs the rule with NashornJS, while this same MIME-type will run the Blocky rule with GraalJS in openHAB 4.
```

File: configuration/blockly/index.md [hunk 2]
```diff
@@ -51,13 +51,43 @@ Using this library you can not only create much simpler code, it also allows **n
...
+There is the choice to
+
+- convert each rule to GraalJS
+- or keep the NashornJS Rules
+
+but both need some work on each blockly rule you have.
+
+#### Migration to GraalJS (recommended)
+
+- Make sure the [JS Scripting Addon](https://www.openhab.org/addons/automation/jsscripting/) addon has been installed.
```
Comment (florian-h05) [inline configuration/blockly/index.md:71]:
```suggestion
- Make sure the [JS Scripting Addon](https://www.openhab.org/addons/automation/jsscripting/) addon is installed.
```

File: configuration/blockly/index.md [hunk 3]
```diff
@@ -51,13 +51,43 @@ Using this library you can not only create much simpler code, it also allows **n
...
+- **To convert / migrate a rule that was created in openHAB 3 (NashornJS) to a compatible rule based on GraalJS for openHAB 4, simply open each Blockly rule once in openHAB 4 and save it - that's it.**
```
Comment (florian-h05) [inline configuration/blockly/index.md:72]:
```suggestion
- **To convert / migrate a rule that was created in openHAB 3 (NashornJS) to a GraalJS-compatible one for openHAB 4, simply open each Blockly rule once in openHAB 4 and save it - that's it.**
```

File: configuration/blockly/index.md [hunk 4]
```diff
@@ -51,13 +51,43 @@ Using this library you can not only create much simpler code, it also allows **n
...
+- In this case, nothing more need to be installed additionally to openHAB 4.
```
Comment (florian-h05) [inline configuration/blockly/index.md:73]:
```suggestion
- In this case, nothing more needs to be installed additionally to openHAB 4.
```

?

File: configuration/blockly/index.md [hunk 5]
```diff
@@ -51,13 +51,43 @@ Using this library you can not only create much simpler code, it also allows **n
...
+
+#### Running openHAB 3 Blockly rules without migrating them right away
+
+- If you still want to run the Blockly rules that were created in openHAB 3 for the time being without changing them (see above), you have to install the  [JavaScript Scripting (Nashorn) Addon](https://www.openhab.org/addons/automation/jsscriptingnashorn/) which could provide backwards compatibility until you have converted all rules.
```
Comment (florian-h05) [inline configuration/blockly/index.md:77]:
```suggestion
- If you still want to run the Blockly rules that were created in openHAB 3 for the time being without changing them (see above), you have to install the [JavaScript Scripting (Nashorn) Addon](https://www.openhab.org/addons/automation/jsscriptingnashorn/) which provides backwards compatibility until you have converted all rules.
```

File: configuration/blockly/index.md [hunk 6]
```diff
@@ -51,13 +51,43 @@ Using this library you can not only create much simpler code, it also allows **n
...
+- Open each Blockly rule o go to the Code TAB o Search for type: application/javascript;
```
Comment (Confectrician) [inline configuration/blockly/index.md:78]:
```suggestion
- Open each Blockly rule or go to the Code TAB or Search for type: application/javascript;
```

Comment (florian-h05) [inline configuration/blockly/index.md]:
```suggestion
- Open each Blockly rule or go to the Code tab or Search for type: application/javascript;
```

FYI: The search for MIME type inside the developer sidebar does not work yet, my PR is still WIP.

**Review by Confectrician — APPROVED** [review]
Just one comment.

Thanks. 👍🏼

**Review by florian-h05 — CHANGES_REQUESTED** [review]
Looks good so far, just a few comments.

I however wonder if we want to talk about „rules“ or rather about „scripts“.

**Comment by Confectrician** [comment]
Sorry @florian-h05 i had the page stillopen and it showed your reviwe comments right after the merge. 🙈

File: configuration/blockly/index.md (hunk 1 above)
Comment (stefan-hoehn) [reply configuration/blockly/index.md:58]: To me and the user, these ARE rules. I would rather keep it that way here.

Comment (florian-h05) [reply configuration/blockly/index.md:58]: You can create scripts in the `Script` tab of UI.
And: In real rules (I mean inside the Rules tab), you add either an action called `Run Script` or `execute a given script` to run something with Blockly.
So IMO these are scripts for the users.

**Pull request merged**